### PR TITLE
clang-tidy.yml: updated to Clang 17

### PR DESCRIPTION
**Pull request by firewave**

### CODE REVIEW

**Comment by firewave** [comment]
I think this project is simple enough that we can keep `misc-include-cleaner` enabled.

**Comment by firewave** [comment]
I filed https://github.com/llvm/llvm-project/issues/64587 about the `misc-include-cleaner` false positive.

File: main.cpp [hunk 1]
```diff
@@ -16,11 +16,16 @@
...
+#include <cstring>
 #include <fstream>
 #include <iostream>
-#include <cstring>
+#include <map>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "simplecpp.h"
```
Comment (danmar) [inline main.cpp:28]: It's an advantage to put local headers before the system includes.

the local header might use types in the system includes and then it's good to enforce that there are corresponding includes in the header.

Comment (firewave) [reply main.cpp:28]: I accidentally did it the wrong around as `simplecpp.h` is a "system" header in the Cppcheck code base - which this is not...

Comment (firewave) [reply main.cpp]: Done.

**Pull request merged**